lib/kcapi-md.c

```diff
@@ -230,6 +230,20 @@ int32_t orig_md_sha512(const uint8_t *in, uint32_t inlen,
 	return (int32_t)kcapi_md_conv_common("sha512", in, inlen, out, outlen);
 }
 
+IMPL_SYMVER(md_sm3, "1.3.1")
+ssize_t impl_md_sm3(const uint8_t *in, size_t inlen,
+		    uint8_t *out, size_t outlen)
+{
+	return kcapi_md_conv_common("sm3", in, inlen, out, outlen);
+}
+
+ORIG_SYMVER(md_sm3, "1.0.0")
+int32_t orig_md_sm3(const uint8_t *in, uint32_t inlen,
+		    uint8_t *out, uint32_t outlen)
+{
+	return (int32_t)kcapi_md_conv_common("sm3", in, inlen, out, outlen);
+}
+
```

Comment:
Since this is a new symbol, you don't need to provide the compat version. I believe you only need the first function with header `ssize_t kcapi_md_sm3(const uint8_t *in, size_t inlen, uint8_t *out, size_t outlen)` (without the `*_SYMVER()`) and add the new symbol into a new block in `lib/version.lds` (consult with @smuellerDD what the version should be for the new block - probably one of `1.3.2` or `1.4.0`).